### ISSUE TITLE
App freezing / hanging?

### ISSUE THREAD

**Issue by guidedways** [DevMate/DevMateKit#36]
Ever since we've incorporated DevMateKit, a number of users have said the app is freezing and hanging for them on 10.10 / 10.11. Coincidence or a known issue?

**Comment by dimaty** [comment]
Yes, we have a few reports about strange hanging when working with activation window. Looking for the real problem with it. Is it the same problem with activation window or an app just starts hanging at any time?

**Comment by guidedways** [comment]
Unfortunately all the users complaining say the main app window does not show and it hangs forever, indefinitely. Strangely if you turn of WiFi and network and then launch, it launches. And turning it back on afterwards makes no difference. 

One user said it happens only after activation is activated. Before when it was in trial it does not happen.

**Comment by dimaty** [comment]
Ok, I'll try to reprodure it and find out the problem. Could you generate and send me 1 day or 5 days activation key to dimaty@devmate.com?

**Comment by guidedways** [comment]
@dimaty thanks again for your help. I've emailed you an activation key along with the app's download link. It's very easy to reproduce this on macOS 10.10 and 10.11 (maybe 10.12 as well, I haven't tested it yet)

1) Launch the app the first time - launches fine. 
2) Quit and relaunch to confirm, no hang.
3) Now launch and enter activation code.
4) Quit and try and relaunch. App hangs forever.

Sampling it using Activity Monitor shows it's hung on some URLConnection request.

5) Turn wifi / network off and launch the app, it launches fine.

**Comment by guidedways** [comment]
Just reading through the issues, I can see a similar issue reported here:

https://github.com/DevMate/DevMateKit/issues/28

The symptoms match and the workaround is to turn wifi off before launching the app (which isn't a workaround).

I've got several upset users at the moment, would appreciate a quick fix. Could be something as trivial as a synchronous network call on the main thread.

This also appears to be an issue simply due to the fact that DevMateKit is linked to the app. The non-direct version of our app has no issues on 10.10 / 10.11

**Comment by guidedways** [comment]
@dimaty any luck on this? I really feel uncomfortable with the hack of a fix I've had to do. Thanks.

**Comment by dimaty** [comment]
Unfortunately I could not reproduce this issue with test app. I've built the app with different versions of Swift and use the same Xcode version as your app. All I can do at this moment is to move "fix" described in issue #28 inside kevlar lib.

**Comment by guidedways** [comment]
> I've built the app with different versions of Swift and use the same Xcode version as your app...

I think this is where the issue is. Our apps aren't made in Swift, they're Obj-C apps. It could be that this issue does not manifest itself with the Swift runtime, so please try the same in Obj-C.

Also, this only occurs when running on 10.10 and 10.11. Did you try it there?

**Comment by ahmaniea** [comment]
We have a similar issue. Some customers are complaining that after entering the activation key, the spinning progress indicator just keeps spinning forever in the activation window. These cases have eventually been solved by first activating the app with a version that had older DevMateKit. We are not able to regenerate this ourselves. Our app is Obj-C app using the latest DevMateKit. 

Don't know if this is related, but at least some of the users had used the 7-day trial with an app version using old DevMateKit. Then the failed activation was made with a version using the latest DevMateKit.
